### PR TITLE
Document (and add!) new flags for 1.0.0

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:
Adds documentation for new Ingress API version and class matching flags.
Adds several flags that I forgot to add everywhere earlier :facepalm: 

**Special notes for your reviewer**:
Whoops, forgot to touch all the flag instances in flags.go in #843 :facepalm:

### CODE REVIEW

**Comment by rainest** [comment]
Was originally going to rebase this to main or open up a separate PR with only the commit that adds the defaults, but on further review, while this is still a goof, it shouldn't matter. Everything in question defaults to `false`, which is effectively what viper defaults to anyway when it can't find a boolean flag. We already get these in https://github.com/Kong/kubernetes-ingress-controller/blob/0.10.0/cli/ingress-controller/flags.go#L346, so...

https://github.com/spf13/viper/blob/v1.7.1/viper.go#L797-L799

https://github.com/spf13/cast/blob/v1.3.1/cast.go#L12-L15
https://github.com/spf13/cast/blob/v1.3.1/caste.go#L80-L81

No one was ever the wiser!

So still :woman_facepalming: but doesn't warrant a patch release to fix the issue in 0.10.x.

File: docs/references/cli-arguments.md [hunk 1]
```diff
@@ -59,6 +62,9 @@ Following table describes all the flags that are available:
 | --profiling                          |`boolean`  | `true`                          | Enable profiling via web interface `host:port/debug/pprof/`. |
 | --publish-service                    |`string`   | none                            | The namespaces and name of the Kubernetes Service fronting Kong Ingress Controller in the form of namespace/name. The controller will set the status of the Ingress resouces to match the endpoints of this service. In reference deployments, this is kong/kong-proxy.|
 | --publish-status-address             |`string`   | none                            | User customized address to be set in the status of ingress resources. The controller will set the endpoint records on the ingress using this address.|
+| --skip-classless-ingress-v1beta1     |`boolean`  | `false`                         | Toggles whether the controller processes `extensions/v1beta1` and `networking/v1beta1` Ingress resources that have no `kubernetes.io/ingress.class` annotation.|
```
Comment (hbagdi) [inline docs/references/cli-arguments.md:65]: This should be `--process-classless-ingress-v1`

File: docs/references/cli-arguments.md [hunk 2]
```diff
@@ -37,6 +37,9 @@ Following table describes all the flags that are available:
 | --alsologtostderr                    |`boolean`  | `false`                         | Logs are written to standard error as well as to files.|
 | --anonymous-reports                  |`string`   | `true`                          | Send anonymized usage data to help improve Kong.|
 | --apiserver-host                     |`string`   | none                            | The address of the Kubernetes Apiserver to connect to in the format of protocol://address:port, e.g., "http://localhost:8080. If not specified, the assumption is that the binary runs inside a Kubernetes cluster and local discovery is attempted.|
+| --disable-ingress-extensionsv1beta1  |`boolean`  | `false`                         | Disable processing Ingress resources with apiVersion `extensions/v1beta1`.|
```
Comment (mflendrich) [inline docs/references/cli-arguments.md:40]: non-blocking: I don't see a point in maintaining this table separately from pflag flag definitions in the code (and the `--help` resulting from there).

By maintaining it, we're asking for extra maintenance burden, and discrepancy bugs.

Tracked (and the necessity of this table to be discussed with @rainest @hbagdi) in #880

**Review by mflendrich — APPROVED** [review]
LGTM.

This PR wouldn't have been needed in the first place (and discrepancies wouldn't have had a chance to occur) if we had implemented the improvements below:

- #880: flag documentation existing both in flag definitions and in markdown - we're susceptible to discrepancy
- #868: the need to touch several places in the code when introducing a new flag - there's potential for accidental omission

**Review by hbagdi — APPROVED** [review]

**Pull request merged**